JavaScript/tic-tac-toe/src/components/TicTacToe.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+import GridButton from "./GridButton";
+import styles from "../styles/tictactoe.module.css"
+
+/** TicTacToe game */
+export default function TicTacToe() {
+    const [gameBoard, setGameBoard] = useState<number[]>(new Array(9).fill(0))
+    const [currentPlayer, setCurrentPlayer] = useState(1)
+    const [playerHasWon, setPlayerHasWon] = useState(false)
+    
+
+    const game = (function() {
```

Comment:
This is a pretty interesting solution to the problem. Not sure that I would have thought to use a "generator" function like this. It's honestly rather elegant.